src/Components/Header.js

```diff
@@ -0,0 +1,51 @@
+import "../Styles/header.css";
+import { AiOutlineSearch } from "react-icons/ai";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { BASE_URL } from "../apis/api";
+
+const Header = ({ handleOpenModal, setImages }) => {
+  const [searchText, setSearchText] = useState("");
+  useEffect(() => {
+    const searchResult = async () => {
+      try {
+        const { data } = await axios.get(`${BASE_URL}search/${searchText}`);
+        setImages(data.result);
+      } catch (e) {
+        console.log(e);
+      }
+    };
+    if (searchText) {
+      const timeOutId = setTimeout(() => {
+        searchResult();
+      }, 1000);
```

Comment:
And I am of the opinion that this implementation is incomplete, mainly because you are not checking if the timeout exist and clearing it if it does exist.

Check your devtools, the Network tab, if for every letter in `searchText` the api is called. If it is called every time, your current implementation is incomplete

Reply:
The api call is not made for every letter I input there is a delay (timeout)  before the api call is made

Reply:
I know, I see the timeout.

You never cancel the timeout, that's why I am of the opinion your implementation is incomplete.

Did you do the test I asked?

Reply:
yes i did

Reply:
Alright then